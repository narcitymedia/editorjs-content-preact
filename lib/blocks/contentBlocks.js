export var BlockType;
(function (BlockType) {
    BlockType["Header"] = "header";
    BlockType["Paragraph"] = "paragraph";
    BlockType["List"] = "list";
    BlockType["RawHTML"] = "raw";
    BlockType["Delimiter"] = "delimiter";
})(BlockType || (BlockType = {}));
export var ListStyle;
(function (ListStyle) {
    ListStyle["Ordered"] = "ordered";
    ListStyle["Unordered"] = "unordered";
})(ListStyle || (ListStyle = {}));
