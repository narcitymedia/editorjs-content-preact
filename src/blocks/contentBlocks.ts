export enum BlockType {
    Header = "header",
    Paragraph = "paragraph",
    List = "list",
    RawHTML= "raw",
    Delimiter = "delimiter"
}

export interface ContentBlock {
    /**
     * Represents the type of block
     */
    type: string,

    /**
     * Data associated with the current block
     */
    data: unknown

    /**
     * Uniquely identifies the block within the content
     */
    contentUniqueId: string
}

export interface HeaderBlock extends ContentBlock {
    
}

export interface ParagraphBlock extends ContentBlock {
    data: {
        /**
         * Text contained in the current paragraph. May contain markup.
         */
        text: string
    }
}
