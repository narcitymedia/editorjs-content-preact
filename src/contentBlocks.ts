export interface ContentBlock {
    /**
     * Represents the type of block
     */
    type: string,

    /**
     * Data associated with the current block
     */
    data: any
}

export interface Header extends ContentBlock {
    
}
