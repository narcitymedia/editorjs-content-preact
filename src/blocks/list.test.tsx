import React from 'react';
import * as renderer from 'react-test-renderer';

import List from './list';
import { ListStyle, ListBlock, BlockType } from './contentBlocks';

describe('List', () => {

  describe('unordered lists', () => {
    it('should render an unordered list with no markup', () => {
      const args: ListBlock = {
        data: {
          items: ["Butter", "Chicken", "Eggs"],
          style: ListStyle.Unordered
        },
        type: BlockType.List,
        contentUniqueId: "XXX-XXX",
      };
      const tree = renderer
        .create(<List {...args} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render an unordered list with markup', () => {
      const args: ListBlock = {
        data: {
          items: ["<p>Butter<p>", "<i>Chicken</i>", "<a href='https://www.google.com'>Eggs</a>"],
          style: ListStyle.Unordered
        },
        type: BlockType.List,
        contentUniqueId: "XXX-XXX",
      };
      const tree = renderer
        .create(<List {...args} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('ordered lists', () => {
    it('should render an ordered list with no markup', () => {
      const args: ListBlock = {
        data: {
          items: ["Butter", "Chicken", "Eggs"],
          style: ListStyle.Ordered
        },
        type: BlockType.List,
        contentUniqueId: "XXX-XXX",
      };
      const tree = renderer
        .create(<List {...args} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render an ordered list with markup', () => {
      const args: ListBlock = {
        data: {
          items: ["<p>Butter<p>", "<i>Chicken</i>", "<a href='https://www.google.com'>Eggs</a>"],
          style: ListStyle.Ordered
        },
        type: BlockType.List,
        contentUniqueId: "XXX-XXX",
      };
      const tree = renderer
        .create(<List {...args} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
