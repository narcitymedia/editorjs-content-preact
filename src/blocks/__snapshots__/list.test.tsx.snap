// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List ordered lists should render an ordered list with markup 1`] = `
<ol>
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Butter<p>",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<i>Chicken</i>",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<a href='https://www.google.com'>Eggs</a>",
      }
    }
  />
</ol>
`;

exports[`List ordered lists should render an ordered list with no markup 1`] = `
<ol>
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Butter",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Chicken",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Eggs",
      }
    }
  />
</ol>
`;

exports[`List unordered lists should render an unordered list with markup 1`] = `
<ul>
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>Butter<p>",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<i>Chicken</i>",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "<a href='https://www.google.com'>Eggs</a>",
      }
    }
  />
</ul>
`;

exports[`List unordered lists should render an unordered list with no markup 1`] = `
<ul>
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Butter",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Chicken",
      }
    }
  />
  <li
    dangerouslySetInnerHTML={
      Object {
        "__html": "Eggs",
      }
    }
  />
</ul>
`;
